File: BackEnd/src/main/java/com/example/pifinance_back/Entities/Topic.java
package com.example.pifinance_back.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Topic implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idTopic;

    private String title;
    private String question;
    private Integer likes;
    private Integer dislikes;

    @Temporal(TemporalType.TIMESTAMP)
    private Date creationDate;
    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private Client user;
    @OneToMany(mappedBy = "topic", cascade = CascadeType.ALL)
    @JsonIgnore
    private Set<Post> posts;
}

File: FrontOffice/src/app/entities/Topic.ts
import { Post } from "./Post";
import { Client } from "./Client";

export class Topic {
    id_topic: any;
    title: any;
    question: any;
    likes: any;
    dislikes: any;
    creation_date: Date;
    user_id: any;
    user: Client;
    posts: Post[];
  }
  
  